site/admin/theatres/edit.php

```diff
@@ -0,0 +1,151 @@
+<?php
+include_once('init.php');
+PrintHeader('Edit Theatre', 'Edit Theatre');
+
+
+echo"<div class='ContactForm'>
+	<form action='' method='get'>
+	Which theatre do you want to edit?
+	<br /><br />";
+@TextBox('EditTheatre');
+echo"<br /><br />
+	<input type='submit' name='EditTheatreSubmitted' />
+	</form>
+	</div>";
+
+if(isset($_REQUEST['EditTheatreSubmitted'])){
+	if(!@$_REQUEST['EditTheatre']){
+		$FormErrors['EditTheatre'] = "Please enter a theatre to edit.";
+	}
+	if(sizeof($FormErrors) == 0){
+		$TheatreInfo = GetTheatre($_REQUEST['EditTheatre']);
+		//echo"".$TheatreInfo['Email']."";
+		//var_dump($TheatreInfo);
+		foreach($TheatreInfo as $Row){
+			echo"<div class='ContactForm'>
+				<h3>Edit the theatre information here:</h3>
+				<br />
+				<form action='' method='post'>
+				Theatre Id:
+			<br /><br />";
+			@ReadOnlyTextBox($Row['TheatreId'], 'TheatreId');
+			echo"<br /><br />
+					Theatre Name:
+				<br /><br />";
+			@EditTextBox($Row['TheatreName'], 'TheatreName');
+			echo"<br /><br />
+					Theatre Description:
+				<br /><br />";
+			@EditTextArea($Row['TheatreDescription'], 'TheatreDescription');
+			echo"<br /><br />
+					Email:
+				<br /><br />";
+			@EditTextBox($Row['Email'], 'Email');
+			echo"<br /><br />
+					Street Address:
+				<br /><br />";
+			@EditTextBox($Row['StreetAddress'], 'StreetAddress');
+			echo"<br /><br />
+					City:
+				<br /><br />";
+			@EditTextBox($Row['City'], 'City');
+			echo"<br /><br />
+					State:
+				<br /><br />";
+			@EditDropDown($Row['State'], 'State', 'Missouri', 'Illinois');
+			echo"<br /><br />
+					Zip:
+				<br /><br />";
+			@EditTextBox($Row['Zip'], 'Zip');
+			echo"<br /><br />
+					Website:
+				<br /><br />";
+			@EditTextBox($Row['Website'], 'Website');
+			echo"<br /><br />
+					Date Formed:
+				<br /><br />";
+			@EditDateInput($Row['DateFormed'], 'DateFormed');
+			echo"<br /><br />
+					Type of theatre:
+				<br /><br />";
+			@EditDropDown($Row['Type'],'Type', 'Equity', 'SPT', 'Non-Equity',
+			'Community');
+			echo"<br /><br />
+					<input type='submit' name='TheatreFormSubmitted' />
+				</form>
+				</div>";
+		}
+	}
+}
+
+if(isset($_REQUEST['TheatreFormSubmitted'])){
```

Comment:
This should go at the top of the file before anything is echoed. In this case it's an even bigger deal than the one I commented on earlier, because if you're validating the form but you'd already echoed the form, none of these errors will be set.